app/request.go

```diff
@@ -59,15 +86,122 @@ func NewRequestCommand(getRequestClientFn GetRequestClientFn) (CommandOut, error
 			Aliases: []string{"g"},
 			Flags: []cli.Flag{
 				&cli.StringFlag{
-					Name:     "request-id",
-					Usage:    "The request-id of the asynchronous request",
+					Name:     RequestIDFlagName,
+					Usage:    "The request-id of the request",
 					Aliases:  []string{"r"},
 					Required: true,
 				},
 			},
 			Action: func(ctx *cli.Context) error {
-				return c.getRequestStatus(ctx.String("request-id"))
+				res, err := c.getRequestStatus(ctx, ctx.String(RequestIDFlagName))
+				if err != nil {
+					return err
+				}
+				return PrintProto(res)
+			},
+		}, {
+			Name:    "wait",
+			Usage:   "wait for the request complete",
+			Aliases: []string{"w"},
+			Flags: []cli.Flag{
+				&cli.StringFlag{
+					Name:     RequestIDFlagName,
+					Usage:    "the request-id of the request",
+					Aliases:  []string{"r"},
+					Required: true,
+				},
+			},
+			Action: func(ctx *cli.Context) error {
+				return c.waitOnRequest(ctx, "", ctx.String(RequestIDFlagName))
 			},
 		}},
 	}}, nil
 }
+
+func (c *RequestClient) waitOnRequest(ctx *cli.Context, operation string, requestID string) error {
+
+	ticker := time.NewTicker(time.Millisecond)
+	defer ticker.Stop()
+	timer := time.NewTimer(ctx.Duration(RequestTimeoutFlagName))
+	defer timer.Stop()
+
+	writer := uilive.New()
+	writer.Start()
+	defer writer.Stop()
+
+	var status *request.RequestStatus
+	defer func() {
+		if status != nil {
+			if err := PrintProto(status); err != nil {
+				fmt.Fprintf(writer, "failed to print status: %s", err)
+			}
+		}
+	}()
+loop:
+	for {
+		select {
+		case <-timer.C:
+			return fmt.Errorf("timed out waiting for request to complete, namespace=%s, requestID=%s, timeout=%s",
+				ctx.String(NamespaceFlagName),
+				requestID,
+				ctx.Duration(RequestTimeoutFlagName),
+			)
+		case <-ticker.C:
+			var err error
+			status, err = c.getRequestStatus(ctx, requestID)
+			if err != nil {
+				return err
+			}
+			switch status.State {
+			case request.STATE_FULFILLED:
+				break loop
+			case request.STATE_FAILED:
+				fmt.Fprintf(writer, "operation failed \n")
+				return fmt.Errorf("request failed: %s", status.FailureReason)
+			case request.STATE_CANCELLED:
+				fmt.Fprintf(writer, "operation cancelled\n")
+				return fmt.Errorf("request was cancelled: %s", status.FailureReason)
+			}
+			if operation != "" {
+				fmt.Fprintf(writer, "waiting for %s operation (requestId='%s') to finish, current state: %s\n",
```

Comment:
`id='%s'` to keep it consistent with the other log statements?

Reply:
fixed.